### PR TITLE
Notebook Snapshotting to the default Notebook isn't working

### PR DESCRIPTION
**Pull request by nikhilmandlik**
### Resolves:
resolves: #4469

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/notebook/components/Notebook.vue [hunk 1]
```diff
@@ -633,15 +628,16 @@ export default {
...
             this.showNav = !this.showNav;
         },
-        async updateDefaultNotebook(notebookStorage) {
-            const defaultNotebookObject = await this.getDefaultNotebookObject();
-            const isSameNotebook = defaultNotebookObject
-                && objectUtils.makeKeyString(defaultNotebookObject.identifier) === objectUtils.makeKeyString(notebookStorage.identifier);
+        updateDefaultNotebook(notebookStorage) {
+            const defaultNotebook = getDefaultNotebook();
+            const defaultNotebookIdentifier = defaultNotebook && defaultNotebook.identifier;
+            const isSameNotebook = defaultNotebookIdentifier
```
Comment (akhenry) [inline src/plugins/notebook/components/Notebook.vue:634]: There is an API function for this - `openmct.object.areIdsEqual(id1, id2)`

File: src/api/objects/ObjectAPI.js [hunk 2]
```diff
@@ -477,7 +481,7 @@ ObjectAPI.prototype.mutate = function (domainObject, path, value) {
 
     if (this.isTransactionActive()) {
         this.transaction.add(domainObject);
-    } else {
+    } else if (!this.isSavingInprogress) {
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:484]: Mutations must be either added to a transaction or _immediately_ saved. 

I think that this change creates a third path there neither of those happen. The mutation _may_ accidentally get persisted by a future mutation to the object, but we can't rely on that.

What was causing the original issue?

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/api/objects/ObjectAPI.js (hunk 2 above)
Comment (nikhilmandlik) [reply src/api/objects/ObjectAPI.js:484]: hmm, inside `save` we call `mutate` and inside `mutate` we call `save`, thats what causing this issue.

**Review by charlesh88 — COMMENTED** [review]
Not sure why I'm a reviewer here: I can't comment on any of the changes in the code.

**Comment by akhenry** [comment]
@nikhilmandlik This looks good. Could you please change the base to `release/1.8.3`? Thank you!

**Review by akhenry — APPROVED** [review]

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? N/A
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Pull request merged**